app/functions/Router.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Functions;
+
+class Router
+{
+  protected $routes = [];
+
+  public function route(string $uri, string $method)
+  {
+    foreach ($this->routes as $route) {
+      if ($route['path'] === $uri && $route['method'] === $method) {
+        $route['function']();
+        return;
+      }
+    }
+    abort(404);
+  }
+
+  protected function addRoute(string $path, string $method, $function): void
+  {
+    $this->routes[] = compact('path', 'method', 'function');
```

Comment:
Myšlienka sa mi páči, ale ak by to bolo možné, rád by som sa pozrel na viac "OOP" formu. teda, čo najmenej používať polia.

Napríklad, dala by sa vytvoriť classa `Route`, kt. by v sebe mala potrebné properties, a ak by to bolo potrebné, neskôr by sme na tom mohli stavať ďalej.

Reply:
Tú triedu Route mám vytvoriť v súbore Router alebo na to vyhradiť nejaký ďalší priečinok a súbor?